src/Wpdb/AbstractBaseWpdbSelectResourceModel.php

```diff
@@ -349,7 +362,41 @@ public function select(
         $limit = null,
         $offset = null
     ) {
-        return $this->_select($condition, $ordering, $limit, $offset);
+        $raw     = $this->_select($condition, $ordering, $limit, $offset);
+        $factory = $this->_getMapFactory();
+        $maps    = [];
+
+        foreach ($raw as $_record) {
+            $maps[] = $factory->make([
```

Comment:
IMO the "making" should be abstracted.

Reply:
Maybe `_createResult()`, so that extending implementations can easily have specialized result objects?

Reply:
Perhaps, abstract twice, so that it's possible to have a specialized result _set_ object as well.